torchgeo/datasets/utils.py

```diff
@@ -266,6 +266,41 @@ def download_radiant_mlhub_collection(
     collection.download(output_dir=download_root, api_key=api_key)
 
 
+def download_azure_container(
+    root: str, name_starts_with: str, *args: Any, **kwargs: Any
+) -> None:
+    """Download a container from Azure blob storage.
+
+    Args:
+        root: Root directory to download to.
+        name_starts_with: Filters the results to return only blobs whose names
+            begin with the specified prefix.
+        *args: Additional arguments to pass to
+            :class:`~azure.storage.blob.ContainerClient`
+        **kwargs: Additional keyword arguments to pass to
+            :class:`~azure.storage.blob.ContainerClient`
+
+    Raises:
+        ImportError: If azure-storage-blob is not installed.
+
+    .. versionadded:: 0.6
+    """
+    try:
+        from azure.storage.blob import ContainerClient
+    except ImportError:
+        raise ImportError(
+            "azure-storage-blob is not installed and is required "
+            "to download this dataset"
+        )
+
+    client = ContainerClient(*args, **kwargs)
+    for blob in client.list_blob_names(name_starts_with=name_starts_with):
```

Comment:
Could wrap this with multiprocessing to speed things up, tqdm to add a progress bar, and various other improvements. There is also an asynchronous version of the API, although I don't think that will benefit us here.

Reply:
TODO: compare download speeds to the CLI version and see if multiprocessing is required to make this comparable.

Reply:
At first glance, this is orders of magnitude slower than `azcopy sync`. I'll try parallelizing it, but if it's still noticeably slower, I suggest we rely on the CLI tool instead. Downside is that it can't be pip install'ed, but being slow is a bigger sin.